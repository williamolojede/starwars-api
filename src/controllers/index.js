export { getMovies } from './getMovies';
export { addComment } from './addComment';
export { getComments } from './getComments';
export { getCharacters } from './getCharacters';
