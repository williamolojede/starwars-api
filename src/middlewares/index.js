export { asyncWrapMiddleware } from './asyncWrapMiddleware';
export { validateRequest } from './validateRequest';
