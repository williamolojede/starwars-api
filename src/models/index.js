import fs from 'fs';
import { Sequelize } from 'sequelize';
import config from '../config/sequelize';

const sequelize = new Sequelize(config.url, config);
const db = {};

fs.readdirSync(__dirname)
  .filter(
    file =>
      file.indexOf('.') !== 0 &&
      file !== 'index.js' &&
      file.slice(-3) === '.js',
  )
  .forEach(file => {
    const model = require(`./${file}`).default.init(sequelize);
    db[model.name] = model;
  });

Object.keys(db).forEach(modelName => {
  if (db[modelName].associate) {
    db[modelName].associate(db);
  }
});

db.sequelize = sequelize;
db.Sequelize = Sequelize;

export default db;

