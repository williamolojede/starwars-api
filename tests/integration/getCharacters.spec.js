import request from 'supertest';

import { app } from '../../src/app';

jest.setTimeout(10000);

describe('Get Characters', () => {
  it('should return a list of characters', async () => {
    const response = await request(app).get('/api/movies/1/characters');
    expect(response.statusCode).toBe(200);
    expect(response.body).toMatchSnapshot();
  });

  it('should return a list of only female characters', async () => {
    const response = await request(app).get('/api/movies/1/characters?filter=female');
    expect(response.statusCode).toBe(200);
    expect(response.body).toMatchSnapshot();
  });

  it('should return a list of characters sorted by name in descending order', async () => {
    const response = await request(app).get('/api/movies/1/characters?sort=name&order=desc');
    expect(response.statusCode).toBe(200);
    expect(response.body).toMatchSnapshot();
  });

  it('should return a list of characters sorted by height in ascending order', async () => {
    const response = await request(app).get('/api/movies/1/characters?sort=height&order=asc');
    expect(response.statusCode).toBe(200);
    expect(response.body).toMatchSnapshot();
  });
});
