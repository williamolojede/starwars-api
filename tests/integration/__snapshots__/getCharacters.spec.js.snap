// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Characters should return a list of characters 1`] = `
Object {
  "data": Array [
    Object {
      "birth_year": "19BBY",
      "created": "2014-12-09T13:50:51.644000Z",
      "edited": "2014-12-20T21:17:56.891000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "male",
      "hair_color": "blond",
      "height": "172",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "77",
      "name": "Luke Skywalker",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/12/",
        "https://swapi.co/api/starships/22/",
      ],
      "url": "https://swapi.co/api/people/1/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/14/",
        "https://swapi.co/api/vehicles/30/",
      ],
    },
    Object {
      "birth_year": "112BBY",
      "created": "2014-12-10T15:10:51.357000Z",
      "edited": "2014-12-20T21:17:50.309000Z",
      "eye_color": "yellow",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/4/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "n/a",
      "hair_color": "n/a",
      "height": "167",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "75",
      "name": "C-3PO",
      "skin_color": "gold",
      "species": Array [
        "https://swapi.co/api/species/2/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/2/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "33BBY",
      "created": "2014-12-10T15:11:50.376000Z",
      "edited": "2014-12-20T21:17:50.311000Z",
      "eye_color": "red",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/4/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "n/a",
      "hair_color": "n/a",
      "height": "96",
      "homeworld": "https://swapi.co/api/planets/8/",
      "mass": "32",
      "name": "R2-D2",
      "skin_color": "white, blue",
      "species": Array [
        "https://swapi.co/api/species/2/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/3/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "41.9BBY",
      "created": "2014-12-10T15:18:20.704000Z",
      "edited": "2014-12-20T21:17:50.313000Z",
      "eye_color": "yellow",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "none",
      "height": "202",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "136",
      "name": "Darth Vader",
      "skin_color": "white",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/13/",
      ],
      "url": "https://swapi.co/api/people/4/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "19BBY",
      "created": "2014-12-10T15:20:09.791000Z",
      "edited": "2014-12-20T21:17:50.315000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "female",
      "hair_color": "brown",
      "height": "150",
      "homeworld": "https://swapi.co/api/planets/2/",
      "mass": "49",
      "name": "Leia Organa",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/5/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/30/",
      ],
    },
    Object {
      "birth_year": "52BBY",
      "created": "2014-12-10T15:52:14.024000Z",
      "edited": "2014-12-20T21:17:50.317000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "brown, grey",
      "height": "178",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "120",
      "name": "Owen Lars",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/6/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "47BBY",
      "created": "2014-12-10T15:53:41.121000Z",
      "edited": "2014-12-20T21:17:50.319000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "female",
      "hair_color": "brown",
      "height": "165",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "75",
      "name": "Beru Whitesun lars",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/7/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "unknown",
      "created": "2014-12-10T15:57:50.959000Z",
      "edited": "2014-12-20T21:17:50.321000Z",
      "eye_color": "red",
      "films": Array [
        "https://swapi.co/api/films/1/",
      ],
      "gender": "n/a",
      "hair_color": "n/a",
      "height": "97",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "32",
      "name": "R5-D4",
      "skin_color": "white, red",
      "species": Array [
        "https://swapi.co/api/species/2/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/8/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "24BBY",
      "created": "2014-12-10T15:59:50.509000Z",
      "edited": "2014-12-20T21:17:50.323000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "black",
      "height": "183",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "84",
      "name": "Biggs Darklighter",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/12/",
      ],
      "url": "https://swapi.co/api/people/9/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "57BBY",
      "created": "2014-12-10T16:16:29.192000Z",
      "edited": "2014-12-20T21:17:50.325000Z",
      "eye_color": "blue-gray",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/4/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "auburn, white",
      "height": "182",
      "homeworld": "https://swapi.co/api/planets/20/",
      "mass": "77",
      "name": "Obi-Wan Kenobi",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/48/",
        "https://swapi.co/api/starships/59/",
        "https://swapi.co/api/starships/64/",
        "https://swapi.co/api/starships/65/",
        "https://swapi.co/api/starships/74/",
      ],
      "url": "https://swapi.co/api/people/10/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/38/",
      ],
    },
    Object {
      "birth_year": "64BBY",
      "created": "2014-12-10T16:26:56.138000Z",
      "edited": "2014-12-20T21:17:50.330000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "auburn, grey",
      "height": "180",
      "homeworld": "https://swapi.co/api/planets/21/",
      "mass": "unknown",
      "name": "Wilhuff Tarkin",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/12/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "200BBY",
      "created": "2014-12-10T16:42:45.066000Z",
      "edited": "2014-12-20T21:17:50.332000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "male",
      "hair_color": "brown",
      "height": "228",
      "homeworld": "https://swapi.co/api/planets/14/",
      "mass": "112",
      "name": "Chewbacca",
      "skin_color": "unknown",
      "species": Array [
        "https://swapi.co/api/species/3/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/10/",
        "https://swapi.co/api/starships/22/",
      ],
      "url": "https://swapi.co/api/people/13/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/19/",
      ],
    },
    Object {
      "birth_year": "29BBY",
      "created": "2014-12-10T16:49:14.582000Z",
      "edited": "2014-12-20T21:17:50.334000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "male",
      "hair_color": "brown",
      "height": "180",
      "homeworld": "https://swapi.co/api/planets/22/",
      "mass": "80",
      "name": "Han Solo",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/10/",
        "https://swapi.co/api/starships/22/",
      ],
      "url": "https://swapi.co/api/people/14/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "44BBY",
      "created": "2014-12-10T17:03:30.334000Z",
      "edited": "2014-12-20T21:17:50.336000Z",
      "eye_color": "black",
      "films": Array [
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "n/a",
      "height": "173",
      "homeworld": "https://swapi.co/api/planets/23/",
      "mass": "74",
      "name": "Greedo",
      "skin_color": "green",
      "species": Array [
        "https://swapi.co/api/species/4/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/15/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "600BBY",
      "created": "2014-12-10T17:11:31.638000Z",
      "edited": "2014-12-20T21:17:50.338000Z",
      "eye_color": "orange",
      "films": Array [
        "https://swapi.co/api/films/4/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "hermaphrodite",
      "hair_color": "n/a",
      "height": "175",
      "homeworld": "https://swapi.co/api/planets/24/",
      "mass": "1,358",
      "name": "Jabba Desilijic Tiure",
      "skin_color": "green-tan, brown",
      "species": Array [
        "https://swapi.co/api/species/5/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/16/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "21BBY",
      "created": "2014-12-12T11:08:06.469000Z",
      "edited": "2014-12-20T21:17:50.341000Z",
      "eye_color": "hazel",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "brown",
      "height": "170",
      "homeworld": "https://swapi.co/api/planets/22/",
      "mass": "77",
      "name": "Wedge Antilles",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/12/",
      ],
      "url": "https://swapi.co/api/people/18/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/14/",
      ],
    },
    Object {
      "birth_year": "unknown",
      "created": "2014-12-12T11:16:56.569000Z",
      "edited": "2014-12-20T21:17:50.343000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "brown",
      "height": "180",
      "homeworld": "https://swapi.co/api/planets/26/",
      "mass": "110",
      "name": "Jek Tono Porkins",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/12/",
      ],
      "url": "https://swapi.co/api/people/19/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "unknown",
      "created": "2014-12-20T19:49:35.583000Z",
      "edited": "2014-12-20T21:17:50.493000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "brown",
      "height": "188",
      "homeworld": "https://swapi.co/api/planets/2/",
      "mass": "79",
      "name": "Raymus Antilles",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/81/",
      "vehicles": Array [],
    },
  ],
  "meta": Object {
    "count": 18,
    "totalHeight": Object {
      "cm": "3066cm",
      "ft_in": "100ft and 7.09inches",
    },
  },
}
`;

exports[`Get Characters should return a list of characters sorted by name in descending order 1`] = `
Object {
  "data": Array [
    Object {
      "birth_year": "64BBY",
      "created": "2014-12-10T16:26:56.138000Z",
      "edited": "2014-12-20T21:17:50.330000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "auburn, grey",
      "height": "180",
      "homeworld": "https://swapi.co/api/planets/21/",
      "mass": "unknown",
      "name": "Wilhuff Tarkin",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/12/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "21BBY",
      "created": "2014-12-12T11:08:06.469000Z",
      "edited": "2014-12-20T21:17:50.341000Z",
      "eye_color": "hazel",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "brown",
      "height": "170",
      "homeworld": "https://swapi.co/api/planets/22/",
      "mass": "77",
      "name": "Wedge Antilles",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/12/",
      ],
      "url": "https://swapi.co/api/people/18/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/14/",
      ],
    },
    Object {
      "birth_year": "unknown",
      "created": "2014-12-20T19:49:35.583000Z",
      "edited": "2014-12-20T21:17:50.493000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "brown",
      "height": "188",
      "homeworld": "https://swapi.co/api/planets/2/",
      "mass": "79",
      "name": "Raymus Antilles",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/81/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "unknown",
      "created": "2014-12-10T15:57:50.959000Z",
      "edited": "2014-12-20T21:17:50.321000Z",
      "eye_color": "red",
      "films": Array [
        "https://swapi.co/api/films/1/",
      ],
      "gender": "n/a",
      "hair_color": "n/a",
      "height": "97",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "32",
      "name": "R5-D4",
      "skin_color": "white, red",
      "species": Array [
        "https://swapi.co/api/species/2/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/8/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "33BBY",
      "created": "2014-12-10T15:11:50.376000Z",
      "edited": "2014-12-20T21:17:50.311000Z",
      "eye_color": "red",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/4/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "n/a",
      "hair_color": "n/a",
      "height": "96",
      "homeworld": "https://swapi.co/api/planets/8/",
      "mass": "32",
      "name": "R2-D2",
      "skin_color": "white, blue",
      "species": Array [
        "https://swapi.co/api/species/2/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/3/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "52BBY",
      "created": "2014-12-10T15:52:14.024000Z",
      "edited": "2014-12-20T21:17:50.317000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "brown, grey",
      "height": "178",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "120",
      "name": "Owen Lars",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/6/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "57BBY",
      "created": "2014-12-10T16:16:29.192000Z",
      "edited": "2014-12-20T21:17:50.325000Z",
      "eye_color": "blue-gray",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/4/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "auburn, white",
      "height": "182",
      "homeworld": "https://swapi.co/api/planets/20/",
      "mass": "77",
      "name": "Obi-Wan Kenobi",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/48/",
        "https://swapi.co/api/starships/59/",
        "https://swapi.co/api/starships/64/",
        "https://swapi.co/api/starships/65/",
        "https://swapi.co/api/starships/74/",
      ],
      "url": "https://swapi.co/api/people/10/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/38/",
      ],
    },
    Object {
      "birth_year": "19BBY",
      "created": "2014-12-09T13:50:51.644000Z",
      "edited": "2014-12-20T21:17:56.891000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "male",
      "hair_color": "blond",
      "height": "172",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "77",
      "name": "Luke Skywalker",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/12/",
        "https://swapi.co/api/starships/22/",
      ],
      "url": "https://swapi.co/api/people/1/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/14/",
        "https://swapi.co/api/vehicles/30/",
      ],
    },
    Object {
      "birth_year": "19BBY",
      "created": "2014-12-10T15:20:09.791000Z",
      "edited": "2014-12-20T21:17:50.315000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "female",
      "hair_color": "brown",
      "height": "150",
      "homeworld": "https://swapi.co/api/planets/2/",
      "mass": "49",
      "name": "Leia Organa",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/5/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/30/",
      ],
    },
    Object {
      "birth_year": "unknown",
      "created": "2014-12-12T11:16:56.569000Z",
      "edited": "2014-12-20T21:17:50.343000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "brown",
      "height": "180",
      "homeworld": "https://swapi.co/api/planets/26/",
      "mass": "110",
      "name": "Jek Tono Porkins",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/12/",
      ],
      "url": "https://swapi.co/api/people/19/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "600BBY",
      "created": "2014-12-10T17:11:31.638000Z",
      "edited": "2014-12-20T21:17:50.338000Z",
      "eye_color": "orange",
      "films": Array [
        "https://swapi.co/api/films/4/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "hermaphrodite",
      "hair_color": "n/a",
      "height": "175",
      "homeworld": "https://swapi.co/api/planets/24/",
      "mass": "1,358",
      "name": "Jabba Desilijic Tiure",
      "skin_color": "green-tan, brown",
      "species": Array [
        "https://swapi.co/api/species/5/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/16/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "29BBY",
      "created": "2014-12-10T16:49:14.582000Z",
      "edited": "2014-12-20T21:17:50.334000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "male",
      "hair_color": "brown",
      "height": "180",
      "homeworld": "https://swapi.co/api/planets/22/",
      "mass": "80",
      "name": "Han Solo",
      "skin_color": "fair",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/10/",
        "https://swapi.co/api/starships/22/",
      ],
      "url": "https://swapi.co/api/people/14/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "44BBY",
      "created": "2014-12-10T17:03:30.334000Z",
      "edited": "2014-12-20T21:17:50.336000Z",
      "eye_color": "black",
      "films": Array [
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "n/a",
      "height": "173",
      "homeworld": "https://swapi.co/api/planets/23/",
      "mass": "74",
      "name": "Greedo",
      "skin_color": "green",
      "species": Array [
        "https://swapi.co/api/species/4/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/15/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "41.9BBY",
      "created": "2014-12-10T15:18:20.704000Z",
      "edited": "2014-12-20T21:17:50.313000Z",
      "eye_color": "yellow",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "none",
      "height": "202",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "136",
      "name": "Darth Vader",
      "skin_color": "white",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/13/",
      ],
      "url": "https://swapi.co/api/people/4/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "200BBY",
      "created": "2014-12-10T16:42:45.066000Z",
      "edited": "2014-12-20T21:17:50.332000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "male",
      "hair_color": "brown",
      "height": "228",
      "homeworld": "https://swapi.co/api/planets/14/",
      "mass": "112",
      "name": "Chewbacca",
      "skin_color": "unknown",
      "species": Array [
        "https://swapi.co/api/species/3/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/10/",
        "https://swapi.co/api/starships/22/",
      ],
      "url": "https://swapi.co/api/people/13/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/19/",
      ],
    },
    Object {
      "birth_year": "112BBY",
      "created": "2014-12-10T15:10:51.357000Z",
      "edited": "2014-12-20T21:17:50.309000Z",
      "eye_color": "yellow",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/4/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "n/a",
      "hair_color": "n/a",
      "height": "167",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "75",
      "name": "C-3PO",
      "skin_color": "gold",
      "species": Array [
        "https://swapi.co/api/species/2/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/2/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "24BBY",
      "created": "2014-12-10T15:59:50.509000Z",
      "edited": "2014-12-20T21:17:50.323000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.co/api/films/1/",
      ],
      "gender": "male",
      "hair_color": "black",
      "height": "183",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "84",
      "name": "Biggs Darklighter",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [
        "https://swapi.co/api/starships/12/",
      ],
      "url": "https://swapi.co/api/people/9/",
      "vehicles": Array [],
    },
    Object {
      "birth_year": "47BBY",
      "created": "2014-12-10T15:53:41.121000Z",
      "edited": "2014-12-20T21:17:50.319000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "female",
      "hair_color": "brown",
      "height": "165",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "75",
      "name": "Beru Whitesun lars",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/7/",
      "vehicles": Array [],
    },
  ],
  "meta": Object {
    "count": 18,
    "totalHeight": Object {
      "cm": "3066cm",
      "ft_in": "100ft and 7.09inches",
    },
  },
}
`;

exports[`Get Characters should return a list of only female characters 1`] = `
Object {
  "data": Array [
    Object {
      "birth_year": "19BBY",
      "created": "2014-12-10T15:20:09.791000Z",
      "edited": "2014-12-20T21:17:50.315000Z",
      "eye_color": "brown",
      "films": Array [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/",
      ],
      "gender": "female",
      "hair_color": "brown",
      "height": "150",
      "homeworld": "https://swapi.co/api/planets/2/",
      "mass": "49",
      "name": "Leia Organa",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/5/",
      "vehicles": Array [
        "https://swapi.co/api/vehicles/30/",
      ],
    },
    Object {
      "birth_year": "47BBY",
      "created": "2014-12-10T15:53:41.121000Z",
      "edited": "2014-12-20T21:17:50.319000Z",
      "eye_color": "blue",
      "films": Array [
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/1/",
      ],
      "gender": "female",
      "hair_color": "brown",
      "height": "165",
      "homeworld": "https://swapi.co/api/planets/1/",
      "mass": "75",
      "name": "Beru Whitesun lars",
      "skin_color": "light",
      "species": Array [
        "https://swapi.co/api/species/1/",
      ],
      "starships": Array [],
      "url": "https://swapi.co/api/people/7/",
      "vehicles": Array [],
    },
  ],
  "meta": Object {
    "count": 2,
    "totalHeight": Object {
      "cm": "315cm",
      "ft_in": "10ft and 4.02inches",
    },
  },
}
`;
